en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml

```diff
@@ -0,0 +1,636 @@
+
+# The schemes supported by the API
+schemes:
+  - https
+
+# The domain of the API.
+# This is configured by the customer during deployment.
+# The given host is just an example.
+host: apis.wso2.com
+
+# The base path of the API.
+# Will be prefixed to all paths.
+basePath: /api/am/gateway/v1
+
+# The following media types can be passed as input in message bodies of the API.
+# The actual media type must be specified in the Content-Type header field of the request.
+# The default is json, i.e. the Content-Type header is not needed to
+# be set, but supporting it serves extensibility.
+consumes:
+  - application/json
+
+# The following media types may be passed as output in message bodies of the API.
+# The media type(s) consumable by the requestor is specified in the Accept header field
+# of the corresponding request.
+# The actual media type returned will be specfied in the Content-Type header field
+# of the of the response.
+# The default of the Accept header is json, i.e. there is not needed to
+# set the value, but supporting it serves extensibility.
+produces:
+  - application/json
+
+securityDefinitions:
+  OAuth2Security:
+    type: oauth2
+    flow: password
+    tokenUrl: https://localhost:9443/oauth2/token
+    scopes:
+      apim:api_publish: Publish API
+
+######################################################
+# The "API Collection" resource APIs
+######################################################
+paths:
+  /deploy-api:
+
+    #-----------------------------------------------------
+    # Deploy the API in the Gateway
+    #-----------------------------------------------------
+    post:
+      security:
+        - OAuth2Security:
+            - apim:api_publish
+      x-wso2-curl: "curl -k -H \"Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8\" -X POST \"https://127.0.0.1:9443/api/am/gateway/v1/deploy-api?apiId=890a4f4d-09eb-4877-a323-57f6ce2ed79b&apiName=PizzaAPI&label=Food\""
+      x-wso2-request: |
+        POST https://127.0.0.1:9443/api/am/gateway/v1/deploy-api?apiId=890a4f4d-09eb-4877-a323-57f6ce2ed79b&apiName=PizzaAPI&label=Food
+        Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8
+      x-wso2-response: "HTTP/1.1 200 OK"
+      summary: Deploy the API in the gateway
+      description: |
+        This operation is used to deploy and API in the gateway.
+      parameters:
+        - name: apiName
+          description: |
+            Name of the API
+          in: query
+          type: string
+          required: true
+        - name: label
+          description: |
+            Label of the API Gateway
+          in: query
+          type: string
+          required: true
+        - $ref: '#/parameters/apiId-Q'
+
+      tags:
+        - Deploy API
+      responses:
+        200:
+          description: |
+            OK.
+            API successfully deployed in the Gateway.
+          schema:
+            $ref: '#/definitions/DeployResponse'
+        202:
```

Comment:
Have we handled these error codes from impl?

Reply:
Handled in rest.api.util